iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/patent/input/PatentReferenceExtractionInputDocumentTextAuthorMerger.java

```diff
@@ -0,0 +1,82 @@
+package eu.dnetlib.iis.wf.referenceextraction.patent.input;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import eu.dnetlib.iis.common.java.io.HdfsUtils;
+import eu.dnetlib.iis.importer.schemas.Author;
+import eu.dnetlib.iis.metadataextraction.schemas.DocumentText;
+import eu.dnetlib.iis.referenceextraction.patent.schemas.DocumentTextWithAuthors;
+import eu.dnetlib.iis.transformers.metadatamerger.schemas.ExtractedDocumentMetadataMergedWithOriginal;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import pl.edu.icm.sparkutils.avro.SparkAvroLoader;
+import pl.edu.icm.sparkutils.avro.SparkAvroSaver;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class PatentReferenceExtractionInputDocumentTextAuthorMerger {
+    private static final SparkAvroLoader sparkAvroLoader = new SparkAvroLoader();
+    private static final SparkAvroSaver sparkAvroSaver = new SparkAvroSaver();
+
+    public static void main(String[] args) throws IOException {
+        JobParameters params = new JobParameters();
+        JCommander jcommander = new JCommander(params);
+        jcommander.parse(args);
+
+        SparkConf conf = new SparkConf();
+        conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
+        conf.set("spark.kryo.registrator", "pl.edu.icm.sparkutils.avro.AvroCompatibleKryoRegistrator");
+
+        try (JavaSparkContext sc = new JavaSparkContext(conf)) {
+            HdfsUtils.remove(sc.hadoopConfiguration(), params.outputPath);
+
+            JavaRDD<DocumentText> documentTexts = sparkAvroLoader.loadJavaRDD(sc, params.inputDocumentTextPath, DocumentText.class);
+            JavaRDD<ExtractedDocumentMetadataMergedWithOriginal> documentMetadatas =
+                    sparkAvroLoader.loadJavaRDD(sc, params.inputExtractedDocumentMetadataMergedWithOriginalPath, ExtractedDocumentMetadataMergedWithOriginal.class);
+
+            JavaPairRDD<CharSequence, DocumentText> documentTextsById = documentTexts
+                    .mapToPair(x -> new Tuple2<>(x.getId(), x));
+
+            JavaPairRDD<CharSequence, List<CharSequence>> documentMetadatasById = documentMetadatas
+                    .filter(x -> Objects.nonNull(x.getImportedAuthors()))
+                    .filter(x -> x.getImportedAuthors().stream().allMatch(y -> StringUtils.isNotBlank(y.getFullname())))
```

Comment:
Do I get it right this `filter` operation refers to a document records not author records? If this is true, wouldn't it exclude documents having only some authors with blank names (and some having full name properly set)?

Reply:
What I mean is: we should accept metadata records if at least one author full name was defined.

Reply:
Changed mapping/filtering to process authors using this schema
  - when `importedAuthors` is `null` -> map to empty list
  - when `importedAuthors` is an empty list -> forward as empty list
  - when `importedAuthors` has authors with `fullname` as `null` -> filter out authors with `null` fullname, include others

I've also added test data to cover these scenarios.